components/Base/Tooltip/index.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+  <div v-if="isDropdown" id="tooltip" class="tooltip">
+    <div class="tooltip__content">
+      <div class="tooltip__content-bar">
+        <div
+          :class="{
+            'tooltip__content-bar--default' : true,
+            'tooltip__content-bar--background-low' : level === 'low',
+            'tooltip__content-bar--background-medium' : level === 'medium',
+            'tooltip__content-bar--background-strong' : level === 'strong'
+          }"
+          v-bind="{...$props,...$attrs}"
+        />
+        <div
+          :class="{
+            'tooltip__content-bar--default' : true,
+            'tooltip__content-bar--background-medium' : level === 'medium',
+            'tooltip__content-bar--background-strong' : level === 'strong'
+          }"
+          v-bind="{...$props,...$attrs}"
+        />
+        <div
+          :class="{
+            'tooltip__content-bar--default' : true,
+            'tooltip__content-bar--background-strong' : level === 'strong'
+          }"
+          v-bind="{...$props,...$attrs}"
+        />
+      </div>
+      <div class="tooltip__content-text">
+        Kata Sandi Anda
+        <span
+          v-if="isLevel"
+          :class="{
+            'tooltip__content-text--weight' : true,
+            'tooltip__content-text--color-low' : level === 'low',
+            'tooltip__content-text--color-medium' : level === 'medium',
+            'tooltip__content-text--color-strong' : level === 'strong',
+          }"
+          v-bind="{...$props,...$attrs}"
+        >
+          {{ textName }}
+        </span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  inheritAttrs: false,
+  props: {
+    level: {
+      type: String,
+      default: 'default'
+    },
```

Comment:
is this `level` props only accepts `low, medium, strong` and `default` values, it's better to add a validator so other developers using this component will know what values are acceptable for this prop

example:
```js
 level: {
      validator: function (value) {
        // The value must match one of these strings
        return ['low', 'medium', 'strong', 'default'].includes(value)
      }
```

ref: https://v2.vuejs.org/v2/guide/components-props.html#Prop-Validation

Reply:
Interesting, But how about the prop validation fails kak?

Reply:
the component will use the set value but will show a warning message in the console, CMIIW @doohanas

Reply:
Yes, I try and read in the docs will show a warning message in the console.

Reply:
The props validator won't help us significantly since the code still runs during a runtime. But it's still a good practice to use the validator, it'll help us during debugging or act as documentation which value should developer set in order to use the component.

Reply:
You can see the changes in [here ](https://github.com/jabardigitalservice/dedi-landing-page/pull/74/commits/80a3a2c64032626051febf9f889b3cd23387fd75) kak